compiler/rustc_const_eval/src/interpret/place.rs

```diff
@@ -115,6 +115,12 @@ impl<'tcx, Tag: Provenance> std::ops::Deref for MPlaceTy<'tcx, Tag> {
     }
 }
 
+impl<'tcx, Tag: Provenance> std::ops::DerefMut for MPlaceTy<'tcx, Tag> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.mplace
```

Comment:
Not having mutable access to these fields was a deliberate decision. Why is this needed?

Reply:
I think this is just leftover from previous refactorings. I missed that one :/

Reply:
this is currently made use of here: https://github.com/b-naber/rust/blob/ef5f07256cfa4e1f43a18acb45e0d8108f8471a4/compiler/rustc_const_eval/src/const_eval/valtrees.rs#L227

Reply:
I think it'd be better to construct a new `MPlaceTy` than change the existing one.